architecture/pagopa.md

```diff
@@ -0,0 +1,203 @@
+
+  1. The source IPs of the connections from `DC-BE` to `PPA-BE` must be static.
+  1. The destination IPs of the connections from `PPA-BE` to `DC-BE` are static.
+
+## Authentication requirements
+
+From _Specifiche di connessione v1.0.2_:
+
+### By PPA-BE
+
+  1. Authentication happens at two layers (L2 and L3).
+  1. At the network layer (L2), the client gets authenticated via the source IP
+     of the connection.
+  1. At the protocol layer (L3), the client gets authenticated via the TLS
+     certificate used by the HTTPS request.
+  1. The HTTPS connections must follow TLS v1.2 or later
+
+#### Requirements for the TLS certificate
+
+  1. Must be of type _x.509 v3 Extended Validation_
+  1. Can be self signed
+  1. The _Subject_ field must be equal to the FQDN of `DC-BE`
+  1. Must be uploaded on the subscription portal or sent via PEC to AgID
+
+### By DC-BE
+
+  1. TLS connections coming from `PPA-BE` will use the same certificate based
+     authentication scheme
+  1. The `PPA-BE` certificate will be provided during the onboarding process
+
+## Applicative entities
+
+From _Specifiche Attuative Nodo v2.1_:
+
+  1. Richiesta Pagamento Telematico (`RPT`) che è emessa dall’Ente Creditore e
+     definisce gli elementi necessari a caratterizzare il pagamento da effettuare;
+  1. Ricevuta Telematica (`RT`), restituita da un PSP a fronte di pagamento
+     individuato da una `RPT`, che definisce gli elementi necessari a qualificare
+     l’esito del pagamento richiesto;
+  1. Richiesta Revoca (`RR`) che è emessa dall’ente interessato e serve a
+     chiedere alla controparte la revoca di una `RT` o lo storno di un pagamento;
+  1. Esito Revoca (`ER`) che è emessa dall’ente cui è stata inviata una `RT` e
+     serve a fornire alla controparte l’esito della revoca di una `RT` o dello
+     storno di un pagamento.
+
+All the above objects are identified by a composite ID made of:
+
+  * codice fiscale dell’Ente Creditore che invia la `RPT`
+  * codice Identificativo Univoco di Versamento (`IUV`)
+
+### RPT
+
+Ogni RPT e la relativa RT possono contenere rispettivamente le informazioni
+attinenti da 1 a 5 pagamenti / esiti a favore di uno stesso Ente Creditore.
+
+#### Notable fields
+
+  1. `identificativoDominio`: Campo alfanumerico contenente il codice fiscale della struttura che invia la richiesta di pagamento
+  1. `codiceIdentificativoUnivoco`: Campo alfanumerico che può contenere il codice fiscale o, in alternativa, la partita IVA del soggetto versante. Nei casi applicabili, quando non è possibile identificare fiscalmente il soggetto, può essere utilizzato il valore `ANONIMO`
+  1. `importoTotaleDaVersare`: Campo numerico (due cifre per la parte decimale, il separatore dei centesimi è il punto `.`), indicante l’importo relativo alla somma da versare.
+  1. `causaleVersamento`
+
+### Avviso push
+
+From _Specifiche Attuative Nodo v2.1_, section 8.1.6.3 (_Processo di avvisatura in modalità Web service_).
```

Comment:
It would be interesting to compare what goes in (from "Ente creditore" to "pagopa backend") with what goes out (from "pagopa backend" notification sent to the PSP). I guess pagopa backend does some enrichment of the message with some internal business logic. In case we can "steal" this logic the EC could send the message using the DC API directly.